frontend/smart-kickers-game/src/hooks/useHeatmap.js

```diff
@@ -0,0 +1,38 @@
+import useAxios from 'axios-hooks';
+import config from '../config';
+import { useMemo } from 'react';
+
+const useHeatmap = () => {
+  const [{ data, loading, error }] = useAxios({
+    method: 'get',
+    url: `${config.apiBaseUrl}/stats`,
+  });
+
+  const heatmap = useMemo(() => (data?.Heatmap ? mirrorHeatmap() : []), [data]);
+
+  function mirrorHeatmap() {
+    const heatmapDim = data.Heatmap.length;
+    const array = new Array(heatmapDim).fill(0).map(() => '');
+    let numbersCopy = JSON.parse(JSON.stringify(data.Heatmap));
+
+    // transpose
+    const transpose = (matrix) => {
+      for (let row = 0; row < matrix.length; row++) {
+        for (let column = 0; column < row; column++) {
+          let temp = matrix[row][column];
+          matrix[row][column] = matrix[column][row];
+          matrix[column][row] = temp;
+        }
+      }
+      return matrix;
+    };
+
+    transpose(numbersCopy);
```

Comment:
You use `transpose` function but you don't assign any variable to that, which is something you shouldn't do. Try to avoid situations where you modify existing variables. It's better to create a new one.

Reply:
Done. Assigned to a variable

Reply:
Change was reversed. When transpose function output was assigned to a function, component threw multiple exceptions.